webpack/scenes/Content/Table/ContentTable.js

```diff
@@ -0,0 +1,118 @@
+  const enabledTypesStatus = useSelector(selectContentTypesStatus);
+  const [contentTypeSelected, setContentTypeSelected] = useState('Python Packages');
+  const [searchQuery, updateSearchQuery] = useState('');
+  const [columnHeaders, setColumnHeaders] = useState([]);
+  const [contentTypes, setContentTypes] = useState(null);
+
+  const { results, ...metadata } = response;
+
+  /* eslint-disable react/no-array-index-key */
+  function buildRows(details) {
+    const rows = [];
+    if (details) {
+      columnHeaders.forEach((header, index) =>
+        rows.push(<Td key={index}>{header.getProperty(details)}</Td>));
+    }
+    return rows;
+  }
+
+  useEffect(() => {
+    dispatch(getContentTypes());
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    if (contentTypes) {
+      setColumnHeaders(contentConfig().find(type =>
+        type.names.singular === contentTypes[contentTypeSelected][0]).columnHeaders);
+    }
+  }, [contentTypes, contentTypeSelected]);
+
+  useMemo(() => {
+    if (enabledTypesStatus === STATUS.RESOLVED) {
+      const types = {};
+      contentConfig().forEach((type) => {
+        if (enabledTypesResponse.includes(type.names.singular)) {
+          types[type.names.title] = [type.names.singular, type.names.plural];
+        }
+      });
+      setContentTypes(types);
+    }
+  }, [enabledTypesResponse, enabledTypesStatus]);
+
+  const fetchItems = useCallback(
+    params => getContent(contentTypes[contentTypeSelected][1], params),
+    [contentTypes, contentTypeSelected],
+  );
+
+  const emptySearchTitle = __('No matching content found');
+  const emptySearchBody = __('Try changing your search settings.');
+
+  if (enabledTypesStatus === STATUS.PENDING || !contentTypes) {
+    return <Loading />;
+  }
+
+  return (
+    <TableWrapper
+      {...{
+        metadata,
+        emptySearchTitle,
+        emptySearchBody,
+        searchQuery,
+        updateSearchQuery,
+        error,
+        status,
+        fetchItems,
+      }}
+      variant={TableVariant.compact}
+      autocompleteEndpoint={`/${contentTypes[contentTypeSelected][1]}/auto_complete_search`}
+      emptyContentTitle={`You currently don't have any ${contentTypeSelected}.`}
+      emptyContentBody={`${contentTypeSelected} will appear here when created.`}
+      actionButtons={
+        <SelectableDropdown
+          items={Object.keys(contentTypes)}
+          title={__('Type')}
+          selected={contentTypeSelected}
+          setSelected={setContentTypeSelected}
+          placeholderText={__('Type')}
+          loading={false}
+          error={false}
+        />
+      }
+    >
+      <Thead>
+        <Tr>
+          {columnHeaders.map(col =>
+            <Th key={col.title}>{col.title}</Th>)}
+        </Tr>
+      </Thead>
+      <Tbody>
+        {results?.map(details => (
+          <Tr key={`${details.pulp_id}`}>
+            {buildRows(details)}
```

Comment:
Not all of our content rabls return a pulp_id but I believe all return an id which will also be needed to route to details page. Maybe use details.id instead of pulp_id.